fabtests/functional/rdm_stress.c

```diff
@@ -593,58 +592,280 @@ static int run_child(void)
 
 	for (i = 0; i < ctrl_cnt && !ret; i++) {
 		printf("(%d-%d) rpc op %s\n", myid, id_at_server,
-		       rpc_op_str(ctrl[i].op));
-		ret = ctrl_op[ctrl[i].op](&ctrl[i]);
+		       rpc_op_str(ctrls[i].op));
+		ret = ctrl_op[ctrls[i].op](&ctrls[i]);
 	}
 
 free:
 	ft_free_res();
 	return ret;
 }
 
+static bool get_uint64_val(const char *json_payload, jsmntok_t *token,
+			   uint64_t *val)
+{
+	if (token->type != JSMN_PRIMITIVE)
+		return false;
 
-static struct rpc_ctrl ctrl_array[] = {
-	{op_goodbye},
-	{op_hello},
-	{op_msg_req, 1000},
-	{op_msg_resp},
-	{op_tag_req, 2000},
-	{op_tag_resp},
-	{op_msg_req, 1000000},
-	{op_msg_resp},
-	{op_write_req, 5600000, {12000}},
-	{op_write_resp},
-	{op_read_req, 64000},
-	{op_read_resp},
-	{op_tag_req, 2000000},
-	{op_tag_resp},
-	{op_read_req, 32000},
-	{op_read_resp},
-	{op_write_req, 86000, {6000}},
-	{op_write_resp},
-	{op_read_req, 1000000},
-	{op_read_resp},
-	{op_write_req, 56000, {12000}},
-	{op_write_resp},
-	{op_sleep, 100},
-	{op_exit},
-};
+	return (sscanf(&json_payload[token->start], "%lu", val) == 1);
+}
 
-/* TODO: read and parse control file */
-static int init_rpc(void)
+static bool get_op_enum(const char *json_payload, jsmntok_t *token,
+			uint32_t *op)
 {
-	ctrl = ctrl_array;
-	ctrl_cnt = ARRAY_SIZE(ctrl_array);
+	const char *enum_val_str;
+	size_t enum_val_len;
+
+	if (token->type != JSMN_STRING)
+		return false;
+
+	enum_val_str = &json_payload[token->start];
+	enum_val_len = token->end - token->start;
+
+	if (!strncmp(enum_val_str, "msg_req", enum_val_len)) {
+		*op = op_msg_req;
+		return true;
+	} else if (!strncmp(enum_val_str, "msg_resp", enum_val_len)) {
+		*op = op_msg_resp;
+		return true;
+	} else if (!strncmp(enum_val_str, "tag_req", enum_val_len)) {
+		*op = op_tag_req;
+		return true;
+	} else if (!strncmp(enum_val_str, "tag_resp", enum_val_len)) {
+		*op = op_tag_resp;
+		return true;
+	} else if (!strncmp(enum_val_str, "read_req", enum_val_len)) {
+		*op = op_read_req;
+		return true;
+	} else if (!strncmp(enum_val_str, "read_resp", enum_val_len)) {
+		*op = op_read_resp;
+		return true;
+	} else if (!strncmp(enum_val_str, "write_req", enum_val_len)) {
+		*op = op_write_req;
+		return true;
+	} else if (!strncmp(enum_val_str, "write_resp", enum_val_len)) {
+		*op = op_write_resp;
+		return true;
+	} else if (!strncmp(enum_val_str, "sleep", enum_val_len)) {
+		*op = op_sleep;
+		return true;
+	} else if (!strncmp(enum_val_str, "goodbye", enum_val_len)) {
+		*op = op_goodbye;
+		return true;
+	} else if (!strncmp(enum_val_str, "exit", enum_val_len)) {
+		*op = op_exit;
+		return true;
+	} else if (!strncmp(enum_val_str, "noop", enum_val_len)) {
+		*op = op_noop;
+		return true;
+	} else if (!strncmp(enum_val_str, "hello", enum_val_len)) {
+		*op = op_hello;
+		return true;
+	}
+
+	return false;
+}
+
+static void init_rpc_ctrl(struct rpc_ctrl *ctrl)
+{
+	ctrl->op = 0xffffffff;
```

Comment:
Maybe set to op_last or add an op_invalid to avoid hard-coding a value.

Reply:
yes, that's a better option